src/core/libmaven/mzMassSlicer.cpp

```diff
@@ -237,89 +245,355 @@ void MassSlices::algorithmC(float ppm, float minIntensity, float rtWindow) {
+                                            const float rtTolerance)
+{
+    auto mz = slice->mz;
+    auto mzMin = slice->mzmin;
+    auto mzMax = slice->mzmax;
+    auto rtMin = slice->rtmin;
+    auto rtMax = slice->rtmax;
+    auto comparisonMz = comparisonSlice->mz;
+    auto comparisonMzMin = comparisonSlice->mzmin;
+    auto comparisonMzMax = comparisonSlice->mzmax;
+    auto comparisonRtMin = comparisonSlice->rtmin;
+    auto comparisonRtMax = comparisonSlice->rtmax;
+    auto mzCenter = (mz + comparisonMz) / 2.0f;
+
+    // check to make sure slices are close to each other (or have some
+    // overlap in mz domain); the tolerance is multiplied 10x so as to
+    // include slices that may be further apart but should be merged
+    float massTolerance = 10.0f * massCutoff->massCutoffValue(mzCenter);
+    if (!(abs(mzCenter - mz) <= massTolerance
+          && abs(mzCenter - comparisonMz) <= massTolerance)) {
+        return make_pair(false, false);
+    }
+
+    // check if common RT regions exist between the slices being compared
+    auto commonLowerRt = 0.0f;
+    auto commonUpperRt = 0.0f;
+    if (rtMin <= comparisonRtMin && rtMax >= comparisonRtMax) {
+        commonLowerRt = comparisonRtMin;
+        commonUpperRt = comparisonRtMax;
+    } else if (rtMin >= comparisonRtMin && rtMax <= comparisonRtMax) {
+        commonLowerRt = rtMin;
+        commonUpperRt  = rtMax;
+    } else if (rtMin >= comparisonRtMin && rtMin <= comparisonRtMax) {
+        commonLowerRt = rtMin;
+        commonUpperRt = min(rtMax, comparisonRtMax);
+    } else if (rtMax >= comparisonRtMin && rtMax <= comparisonRtMax) {
+        commonLowerRt = max(rtMin, comparisonRtMin);
+        commonUpperRt = rtMax;
+    }
+    if (commonLowerRt == 0.0f && commonUpperRt == 0.0f)
+        return make_pair(false, true);
+
+    auto highestIntensity = 0.0f;
+    auto mzAtHighestIntensity = 0.0f;
+    auto rtAtHighestIntensity = 0.0f;
+    auto highestCompIntensity = 0.0f;
+    auto mzAtHighestCompIntensity = 0.0f;
+    auto rtAtHighestCompIntensity = 0.0f;
+    for (auto sample : samples) {
+        // obtain EICs for the two slices
+        auto eic = sample->getEIC(mzMin,
+                                  mzMax,
+                                  rtMin,
+                                  rtMax,
+                                  1,
+                                  1,
+                                  "");
+        auto comparisonEic = sample->getEIC(comparisonMzMin,
+                                            comparisonMzMax,
+                                            comparisonRtMin,
+                                            comparisonRtMax,
+                                            1,
+                                            1,
+                                            "");
+
+        // obtain the highest intensity's mz and rt in the EICs
+        highestIntensity = eic->maxIntensity;
```

Comment:
This will always store the attributes for the last sample, right?

Reply:
Right. This is a bug.